packages/contracts-bridge/contracts/test/BridgeRouter.t.sol

```diff
@@ -297,6 +716,266 @@ contract BridgeRouterTest is BridgeTest {
         assertEq(revertingToHook.test(), 123);
     }
 
+    function testFuzz_handleHookTransferSucceeds(
+        uint256 tokenAmount,
+        bytes32 tokenDetailsHash,
+        bytes memory extraData,
+        bytes32 sender,
+        uint32 nonce
+    ) public {
+        tokenAmount = bound(
+            tokenAmount,
+            0,
+            type(uint256).max - bridgeUserTokenAmount
+        );
+        bytes32 hook = address(revertingToHook).addressToBytes32();
+        // The hook will succed only if origin < 10. This has nothing
+        // to do with how transferToHook works, but rather how the mock works.
+        // We created a mock that will have different behaviour depending on the origin,
+        // so that we can test different scenarios easily.
+        uint32 origin = 1;
+        localToken.mint(address(bridgeRouter), tokenAmount);
+        bytes memory action = abi.encodePacked(
+            BridgeMessage.Types.TransferToHook,
+            hook,
+            tokenAmount,
+            tokenDetailsHash,
+            sender,
+            extraData
+        );
+        bytes memory tokenId = abi.encodePacked(
+            localDomain,
+            address(localToken).addressToBytes32()
+        );
+        // The hook succeeds
+        bridgeRouter.exposed_handleTransferToHook(
+            origin,
+            nonce,
+            tokenId,
+            action
+        );
+        assertEq(revertingToHook.test(), 123);
+    }
+
+    event Receive(
+        uint64 indexed originAndNonce,
+        address indexed token,
+        address indexed recipient,
+        address liquidityProvider,
+        uint256 amount
+    );
+
+    function test_dust() public {
+        address alice = address(0xBEEEEF);
+        address bob = address(0xBEEEEEEEEEF);
+        vm.deal(alice, 10 ether);
+        vm.deal(address(bridgeRouter), 20 ether);
+
+        bridgeRouter.exposed_dust(alice);
+        bridgeRouter.exposed_dust(bob);
+
+        assertEq(alice.balance, 10 ether);
+        assertEq(bob.balance, bridgeRouter.DUST_AMOUNT());
+        assertEq(
+            address(bridgeRouter).balance,
+            20 ether - bridgeRouter.DUST_AMOUNT()
+        );
+    }
+
+    function testFuzz_originAndNonce(uint32 origin, uint32 nonce) public {
+        assertEq(
+            bridgeRouter.exposed_originAndNonce(origin, nonce),
+            uint256((uint64(origin) << 32) | nonce)
+        );
+    }
+
+    function test_renounceOwnership() public {
+        address owner = bridgeRouter.owner();
+        bridgeRouter.renounceOwnership();
+        assertEq(bridgeRouter.owner(), owner);
+    }
+}
+
+contract nonEthereumBridgeRouterTest is BridgeRouterFixture {
```

Comment:
`PascalCase`

But really, no need for the `non` at all. Just omit `Ethereum` and call it `BridgeRouterTest` ?

Reply:
should this be in a separate file?